### DISCUSSION TITLE
Three little tools

### DISCUSSION THREAD

**Discussion by alaskamike** [nmalkin/plot-likert#27] [Ideas]
Here are three tools I use to prepare data for plotting. I have them in my personal plot_likert.py file and I think they'd be good to add to the package.

Mike

```
# data prep tools
# run as needed in the order listed here!

#reverse negatative questions
def reverser(df, start_col, end_col):
    for c in range(start_col, end_col):
        for r in range(0, len(df)-1):
            if df.iloc[r, c] > 0:
                df.iloc[r,c] = (8.0 - df.iloc[r,c])
    return(df)

# create short names for questions
def numberer(df, start_col, end_col):
    new_df = pd.DataFrame()
    for q in range(start_col, end_col):
        if q < 9:
            num = '0' + str(q+1)
        else:
            num = str(q+1)
        new_col = 'Q_' + num
        new_df[new_col] = df.iloc[:,q].astype(str).str.slice(0,1).astype(float)
    return(new_df)

# converts integers to strings for plotting
def converter(df):
    #new_df = pd.DataFrame()
    cols = len(df.columns)
    for i in range(0,cols):
        df.iloc[:,i] = df.iloc[:,i].astype(int).astype(str)
    return(df)
```

**Comment by nmalkin** [comment]
Thank you for sharing these snippets! They do seem like they can be quite useful.

I'm a little scared of adding lots of new features to the package right now, because I don't have a proper test suite. (One of my commits broke the build recently, and I didn't pick up on it for almost a month. 😞)

I think another prerequisite might be for the library to get better at handling different-length scales. For instance, I noticed your `reverser` subtracts values from `8.0`, but — as an example — most of the scales in my work have 5 points.

I do agree that the library could be more useful if it included some of these data cleaning features. If I end up finding the time to do that, I'll definitely try to include your tools. In the meantime, thank you for posting them here! Hopefully other users will be able to find them and make use of them in their own projects.

**Comment by alaskamike** [comment]
Good catch on the scale for reverser - I usually have 7-point scales to
deal with. I will update that and resend.

I like having these functions in my personal plot_likert.py file but I
understand that they aren't a fit.  Maybe we could have a
plot_likert_tools.py that is called by plot_likert?
